rfc/helm-synthesis.md

```diff
@@ -0,0 +1,201 @@
+```
+apiVersion: v2                        
+name: <app-directory-name>                                 
+version: 1.0.0                             
+description: Generated chart for <app-directory-name>  
+type: application
+```
+
+With this generated chart, now you can run `helm install <release-name> ./dist` to install the chart to your Kubernetes cluster.
+
+
+> **Note:**
+Templates within the generated Helm chart are pure and static Kubernetes manifests; they don't contain any helm template directives. This means they cannot be customized with a `values.yaml` file or the release name. One important implication of this is that you cannot deploy two releases of the chart, as resource names will collide. If you need customization, you can do this within the cdk8s application (for example by explicitly reading a `values.yaml` file).
+
+---
+
+> Ticking the box below indicates that the public API of this RFC has been signed-off by the API bar raiser (the `api-approved` label was applied to the RFC pull request):
+
+```
+[ ] Signed-off by API Bar Raiser @iliapolo
+```
+
+---
+
+## Public FAQ
+
+### What are we launching today?
+
+We have added a new option to `cdk8s synth` command that allows users to synthesize their apps into helm charts.
+
+### Why should I use this feature?
+
+You should use this feature if you'd like to deploy your cdk8s application using Helm.
+
+---
+
+## Internal FAQ
+
+### Why are we doing this?
+
+Helm is a powerful package manager for Kubernetes and is also popular among developers as a deployment tool. It eases the process of deploying to Kubernetes cluster for its users.
+
+Currently, a cdk8s application produces a directory of pure Kubernetes manifests. Deploying those manifests using helm requires user intervention because helm expect a different directory structure, with custom metadata files.
+
+Now, with the implementation of this feature, the user can run,
+```
+cdk8s synth --format helm --chart-api-version v2 --chart-version 1.0.0 && helm install <release-name> ./dist
+```
+This will create a helm chart containing the generated manifests and deploy them to the Kubernetes cluster via helm. Also note that until now, they only way to **easily** deploy the output of `cdk8s synth` was using `kubectl apply`, which is less common in production workloads. This feature opens the door for robust and easy deployment of cdk8s applications in a production environment. 
+
+### Why should we _not_ do this?
+
+Customers are not currently blocked for deploying to helm with our synthesized manifests. They would need to create the appropriate helm chart folder structure by themselves and place the synthesized manifests in the templates folder of the chart. With this structure, they can deploy the manifests to their Kubernetes cluster. Now, since a workaround already exists, if we proceed with implementing this RFC, this would take up developers time and effort and will add to maintenance load of cdk8s.
```

Comment:
```suggestion
Customers are not currently blocked for deploying to helm with our synthesized manifests. They would need to create the appropriate helm chart folder structure by themselves and place the synthesized manifests in the templates folder of the chart. With this structure, they can deploy the manifests to their Kubernetes cluster. Since a workaround already exists, if we proceed with implementing this RFC, this would take up developers time and effort and will add to maintenance load of cdk8s.
```